kyuubi-server/src/main/antlr4/org/apache/kyuubi/sql/KyuubiSQL.g4

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+grammar KyuubiSQL;
+
+@members {
+   /**
+    * Verify whether current token is a valid decimal token (which contains dot).
+    * Returns true if the character that follows the token is not a digit or letter or underscore.
+    *
+    * For example:
+    * For char stream "2.3", "2." is not a valid decimal token, because it is followed by digit '3'.
+    * For char stream "2.3_", "2.3" is not a valid decimal token, because it is followed by '_'.
+    * For char stream "2.3W", "2.3" is not a valid decimal token, because it is followed by 'W'.
+    * For char stream "12.0D 34.E2+0.12 "  12.0D is a valid decimal token because it is followed
+    * by a space. 34.E2 is a valid decimal token because it is followed by symbol '+'
+    * which is not a digit or letter or underscore.
+    */
+   public boolean isValidDecimal() {
+     int nextChar = _input.LA(1);
+     if (nextChar >= 'A' && nextChar <= 'Z' || nextChar >= '0' && nextChar <= '9' ||
+       nextChar == '_') {
+       return false;
+     } else {
+       return true;
+     }
+   }
+ }
+
+tokens {
+    DELIMITER
+}
+
+singleStatement
+    : statement ';'* EOF
+    ;
+
+statement
+    : runnable                                                      #logicalRunnable
+    | translate                                                     #logicalTranslate
+    | .*?                                                           #passThrough
+    ;
+
+runnable
+    : KILL ENGINE                                                   #killEngine
+    | LAUNCH ENGINE                                                 #launchEngine
+    | ALTER SESSION SET propertyList                                #alterEngineConfig
+    ;
+
+translate
+    : RENAME TABLE identifier TO identifier                         #renameTable
```

Comment:
we should be careful about extending the syntax, currently, the generic Kyuubi SQL scope is narrowed to management. it says, the typical ANSI-like SQL is 

```
SHOW <RESOURCES>
CREATE <RESOURCE> ...
ALTER <RESOURCE> SET ...
DROP <RESOURCE> ...
```

In Kyuubi case, I think we can start w/ the resource `SESSION` or `ENGINE`.


The "translate" should be protocol-specific, e.g. we only need it for MySQL FE now.

Reply:
Thanks for your thoughts, That's great; and thanks @wForget, prefer this style to that of `CALL  func(params)`, what do you think~

@pan3793 I like this style for extending the syntaxm, but and have a bit weird if we use `CREATE ENGINE` for launch engine, `DROP ENGINE` for kill/stop engine.